src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -766,26 +766,18 @@ public void validate()
          throw new IllegalArgumentException("poolName cannot contain ':' when used with JMX");
       }
 
-      if (driverClassName != null && jdbcUrl == null) {
-         LOGGER.error("jdbcUrl is required with driverClassName");
-         throw new IllegalArgumentException("jdbcUrl is required with driverClassName");
+      if (dataSource != null) {
+         LOGGER.debug("{} - using dataSource={}", poolName, dataSource);
       }
-      else if (driverClassName != null && dataSourceClassName != null) {
-         LOGGER.error("cannot use driverClassName and dataSourceClassName together");
-         throw new IllegalArgumentException("cannot use driverClassName and dataSourceClassName together");
-      }
-      else if (jdbcUrl != null && dataSourceClassName != null) {
-         LOGGER.warn("using dataSourceClassName and ignoring jdbcUrl");
```

Comment:
Why would you remove these errors and warnings?  *Nobody* runs with debug logging on.  These are some of the most common misconfigurations and misunderstandings.  Removing these is going to generate *more* support requests.

I'd be happier if you simply threw exceptions instead.